### ISSUE TITLE
[protocol] Add `Search` support.

### ISSUE THREAD

**Issue by ejgallego** [ejgallego/coq-lsp#324]

**Comment by mattam82** [comment]
Any hope to make a quick and dirty search/searchabout available? Even when writting `Search foo` in the document I don't see any result. It's kind of crucial to work on large libraries. And definitely the existing coq-lsp speedup is far more important than Search's performance.

**Comment by Alizter** [comment]
@mattam82 That's strange. Search should show something in the panel. Is your typing cursor over the Search statement?

**Comment by mattam82** [comment]
Ah I see, it was just after the statement

**Comment by Alizter** [comment]
@mattam82 There is an option that shows the messages of the sentence before the cursor which might be useful if you find yourself backtracking alot.

**Comment by ejgallego** [comment]
@mattam82 I fully agree we need better search; I've discussed with some people about plans, but so far no one took on it yet, and my backlog is a bit large these days.

Indeed we can now do a bit better with how the cursor works I think, as always this kind of stuff is subtle.

What happens in Coq now when `Search foo.` is typed is that the sentence will indeed stop at the point.

I think the option @Alizter mentions doesn't affect the display of messages, but only of goals.

**Comment by adrianleh** [comment]
In 0.1.8 for Coq v8.18, I no longer see any results in messages when I use Search/Check/Print. This was not an issue previously on Coq 8.16 with lsp.

**Comment by Alizter** [comment]
@adrianleh Do you see the results when your cursor is over the Search statement or before and after it. There is a configuration option that displays the goal / messages at the cursor vs just before.

**Comment by adrianleh** [comment]
I only see it upon mouse-over, however there is no text cursor state (either before or after the period) where is see it.

**Comment by adrianleh** [comment]
https://github.com/ejgallego/coq-lsp/assets/26521502/0337e755-6bac-46fb-9e06-6f178b33fd15

**Comment by Alizter** [comment]
@adrianleh Hmm that is very strange. Does the same happen when you do a check at the top of the file? Sometimes a hanging proof may stop the goal window from being updated.

I have the same version numbers as you and I haven't seen any such issue when using Search.

**Comment by adrianleh** [comment]
This happens anywhere in the file

**Comment by ejgallego** [comment]
Hi @adrianleh , it seems to me that this is a problem with the extension being activated multiple times, and maybe some configuration mixup.

I think `coq-lsp` Zulip could be a better place to discuss this, would you mind if we move the discussion there?

**Comment by adrianleh** [comment]
Adding https://coq.zulipchat.com/#narrow/stream/329642-coq-lsp/topic/LSP.20search.20issue/near/419497086 as a reference
